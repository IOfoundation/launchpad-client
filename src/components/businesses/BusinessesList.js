import React, {Component} from 'react';
import {PropTypes} from 'prop-types';

import Business from './Business';

//const BusinessesList = ({businesses, handleClickOnBusiness, expanded}) => {
class BusinessesList extends Component {
  render() {
    const {handleClickOnBusiness, organizations} = this.props;
    return (
      <div>
        {organizations.map(organization => {
          return (
            <Business
              business={organization}
              key={organization.id}
              handleClickOnBusiness={handleClickOnBusiness}
            />
          );
        })}
      </div>
    );
  }
}

BusinessesList.propTypes = {
  locations: PropTypes.array.isRequired,
  organizations: PropTypes.array.isRequired,
  expanded: PropTypes.bool,
  handleClickOnBusiness: PropTypes.func.isRequired,
};

export default BusinessesList;
