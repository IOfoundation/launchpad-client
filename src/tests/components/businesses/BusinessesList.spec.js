import React from 'react';
import {shallow} from 'enzyme';

import BusinessesList from '../../../components/businesses/BusinessesList';

const handleClickOnClearAllFilters = jest.fn();

function mockMultipleOrganizations() {
  return [
    {
      id: 1,
      name: 'Businesses1',
    },
    {
      id: 2,
      name: 'Businesses2',
    },
  ];
}

describe('<BusinessesList />', () => {
  it('Renders empty state', () => {
    const organizations = emptyOrganizations();
    const wrapper = shallow(
      <BusinessesList
        organizations={organizations}
        handleClickOnBusiness={handleClickOnBusiness}
        handleClickOnClearAllFilters={handleClickOnClearAllFilters}
      />
    );
    expect(wrapper.find('.no-result-message-show'));
  });

  it('Renders expanded Business component when only one Org in the array', () => {
    const organizations = mockSingleOrganization();
    const wrapper = shallow(
      <BusinessesList
        organizations={organizations}
        handleClickOnBusiness={handleClickOnBusiness}
        handleClickOnClearAllFilters={handleClickOnClearAllFilters}
      />
    );
    expect(wrapper.find('Business').prop('expanded')).toEqual(true);
  });

  it('Renders a Business component for every Org in the array', () => {
    const organizations = mockMultipleOrganizations();
    const isMobile = false;
    const wrapper = shallow(
      <BusinessesList
        organizations={organizations}
        handleClickOnClearAllFilters={handleClickOnClearAllFilters}
        isMobile={isMobile}
      />
    );
    expect(wrapper.find('Business').length).toEqual(2);
  });
});
