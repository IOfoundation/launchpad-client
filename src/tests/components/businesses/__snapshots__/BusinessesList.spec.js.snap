// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BusinessesList /> renders snapshot of BusinessesList -- one org in array 1`] = `
<BusinessesList
  isMobile={false}
  organizations={
    Array [
      Object {
        "contacts": Array [
          Object {
            "email": "jondoe@test.org",
            "id": 1,
            "name": "Jon Doe",
            "phones": Array [
              Object {
                "number": "(530) 795-1520",
              },
            ],
          },
        ],
        "description": "orgDescription",
        "email": "jondoe@something.org",
        "id": 1,
        "locations": Array [
          Object {
            "address": Object {
              "address_1": "123 Road St",
              "address_2": null,
              "city": "Sacramento",
              "postal_code": "12345",
              "state_province": "CA",
            },
            "id": 1,
            "name": "location1",
            "phones": Array [
              Object {
                "id": 2,
                "number": "(123) 456 - 7890",
              },
            ],
          },
          Object {
            "address": Object {
              "address_1": "456 Street Rd",
              "address_2": null,
              "city": "Sacramento",
              "postal_code": "12345",
              "state_province": "CA",
            },
            "id": 2,
            "name": "location2",
            "phones": Array [
              Object {
                "id": 3,
                "number": "(123) 456 - 7890",
              },
            ],
          },
        ],
        "name": "organization1",
        "phones": Array [
          Object {
            "id": 1,
            "number": "(123) 456 - 7890",
          },
        ],
        "services": Array [
          Object {
            "categories": Array [
              Object {
                "id": 54,
                "name": "category1",
              },
              Object {
                "id": 55,
                "name": "category2",
              },
              Object {
                "id": 56,
                "name": "category3",
              },
            ],
            "description": "lorem ipsum...",
            "email": null,
            "id": 1,
            "name": "service1",
            "phones": Array [],
          },
          Object {
            "categories": Array [
              Object {
                "id": 54,
                "name": "category1",
              },
            ],
            "description": "lorem ipsum...",
            "email": null,
            "id": 2,
            "name": "service2",
            "phones": Array [],
          },
        ],
      },
    ]
  }
  selectedOrg={-1}
>
  <div>
    <withRouter(Business)
      business={
        Object {
          "contacts": Array [
            Object {
              "email": "jondoe@test.org",
              "id": 1,
              "name": "Jon Doe",
              "phones": Array [
                Object {
                  "number": "(530) 795-1520",
                },
              ],
            },
          ],
          "description": "orgDescription",
          "email": "jondoe@something.org",
          "id": 1,
          "locations": Array [
            Object {
              "address": Object {
                "address_1": "123 Road St",
                "address_2": null,
                "city": "Sacramento",
                "postal_code": "12345",
                "state_province": "CA",
              },
              "id": 1,
              "name": "location1",
              "phones": Array [
                Object {
                  "id": 2,
                  "number": "(123) 456 - 7890",
                },
              ],
            },
            Object {
              "address": Object {
                "address_1": "456 Street Rd",
                "address_2": null,
                "city": "Sacramento",
                "postal_code": "12345",
                "state_province": "CA",
              },
              "id": 2,
              "name": "location2",
              "phones": Array [
                Object {
                  "id": 3,
                  "number": "(123) 456 - 7890",
                },
              ],
            },
          ],
          "name": "organization1",
          "phones": Array [
            Object {
              "id": 1,
              "number": "(123) 456 - 7890",
            },
          ],
          "services": Array [
            Object {
              "categories": Array [
                Object {
                  "id": 54,
                  "name": "category1",
                },
                Object {
                  "id": 55,
                  "name": "category2",
                },
                Object {
                  "id": 56,
                  "name": "category3",
                },
              ],
              "description": "lorem ipsum...",
              "email": null,
              "id": 1,
              "name": "service1",
              "phones": Array [],
            },
            Object {
              "categories": Array [
                Object {
                  "id": 54,
                  "name": "category1",
                },
              ],
              "description": "lorem ipsum...",
              "email": null,
              "id": 2,
              "name": "service2",
              "phones": Array [],
            },
          ],
        }
      }
      expanded={false}
      isMobile={false}
      isSelected={false}
      key="1"
    >
      <Business
        business={
          Object {
            "contacts": Array [
              Object {
                "email": "jondoe@test.org",
                "id": 1,
                "name": "Jon Doe",
                "phones": Array [
                  Object {
                    "number": "(530) 795-1520",
                  },
                ],
              },
            ],
            "description": "orgDescription",
            "email": "jondoe@something.org",
            "id": 1,
            "locations": Array [
              Object {
                "address": Object {
                  "address_1": "123 Road St",
                  "address_2": null,
                  "city": "Sacramento",
                  "postal_code": "12345",
                  "state_province": "CA",
                },
                "id": 1,
                "name": "location1",
                "phones": Array [
                  Object {
                    "id": 2,
                    "number": "(123) 456 - 7890",
                  },
                ],
              },
              Object {
                "address": Object {
                  "address_1": "456 Street Rd",
                  "address_2": null,
                  "city": "Sacramento",
                  "postal_code": "12345",
                  "state_province": "CA",
                },
                "id": 2,
                "name": "location2",
                "phones": Array [
                  Object {
                    "id": 3,
                    "number": "(123) 456 - 7890",
                  },
                ],
              },
            ],
            "name": "organization1",
            "phones": Array [
              Object {
                "id": 1,
                "number": "(123) 456 - 7890",
              },
            ],
            "services": Array [
              Object {
                "categories": Array [
                  Object {
                    "id": 54,
                    "name": "category1",
                  },
                  Object {
                    "id": 55,
                    "name": "category2",
                  },
                  Object {
                    "id": 56,
                    "name": "category3",
                  },
                ],
                "description": "lorem ipsum...",
                "email": null,
                "id": 1,
                "name": "service1",
                "phones": Array [],
              },
              Object {
                "categories": Array [
                  Object {
                    "id": 54,
                    "name": "category1",
                  },
                ],
                "description": "lorem ipsum...",
                "email": null,
                "id": 2,
                "name": "service2",
                "phones": Array [],
              },
            ],
          }
        }
        expanded={false}
        isMobile={false}
        isSelected={false}
      >
        <div
          className="business-card"
          style={
            Object {
              "backgroundColor": "#F2F2F2",
            }
          }
        >
          <div
            className="business-card-collapse grid"
          >
            <div
              className="business business-img"
              onClick={[Function]}
            >
              <img
                className="business-logo"
                src="../static-data/images/orgs-placeholder.png"
              />
            </div>
            <div
              className="business col-lg-12 col-md-12 col-xs-12 p-0"
            >
              <div
                className="business-details-container"
              >
                <div
                  className="business-information"
                  onClick={[Function]}
                >
                  <h3
                    className="title m-bot-8"
                  >
                    organization1
                  </h3>
                  <p
                    className="business-description"
                  >
                    orgDescription
                  </p>
                </div>
                <div
                  className="social-icons-hide"
                />
              </div>
              <img
                className="business-card-icon"
                onClick={[Function]}
                src="static-data/images/expand-icon.png"
                style={
                  Object {
                    "float": "right",
                  }
                }
              />
            </div>
            <div
              className="grid col-lg-12 col-md-12 col-xs-12 full-information p-0"
            >
              <div
                className="grid col-lg-12 col-md-12 col-xs-12 p-0 m-bot-25"
              >
                <div
                  className="col-lg-4 col-md-4 col-xs-12 p-0 m-right-52 main-location"
                >
                  <p
                    className="business-title"
                  >
                    Main Location:
                  </p>
                  <h4>
                    123 Road St
                  </h4>
                  <h4>
                    Sacramento, CA
                  </h4>
                </div>
                <div
                  className="col-lg-5 col-md-5 col-xs-12 p-0 main-contact"
                >
                  <p
                    className="business-title"
                  >
                    Contact:
                  </p>
                  <h4>
                    (530) 795-1520
                  </h4>
                  <h4
                    className="break-word"
                  >
                    jondoe@test.org
                  </h4>
                </div>
              </div>
              <hr />
              <p
                className="business-title col-lg-12 col-md-12 col-xs-12 p-0"
              >
                Services
              </p>
              <div
                className="col-lg-12 col-md-12 col-xs-12 grid p-0 business-service m-bot-28"
                key="1"
              >
                <div
                  className="service-description col-lg-12 col-md-12 col-xs-12 p-left-0"
                >
                  <h4
                    className="text-bold"
                  >
                    service1
                  </h4>
                  <p>
                    lorem ipsum...
                  </p>
                </div>
                <div
                  className="col-lg-12 col-md-12 col-xs-12 p-0 m-top-16"
                >
                  <Chip
                    canDelete={false}
                    key="54"
                    text="category1"
                  >
                    <a
                      className="
      search-filter-label
    "
                      data-value="category1"
                      onClick={[Function]}
                    >
                      <span
                        className="text"
                      >
                        category1
                      </span>
                    </a>
                  </Chip>
                  <Chip
                    canDelete={false}
                    key="55"
                    text="category2"
                  >
                    <a
                      className="
      search-filter-label
    "
                      data-value="category2"
                      onClick={[Function]}
                    >
                      <span
                        className="text"
                      >
                        category2
                      </span>
                    </a>
                  </Chip>
                  <Chip
                    canDelete={false}
                    key="56"
                    text="category3"
                  >
                    <a
                      className="
      search-filter-label
    "
                      data-value="category3"
                      onClick={[Function]}
                    >
                      <span
                        className="text"
                      >
                        category3
                      </span>
                    </a>
                  </Chip>
                </div>
              </div>
              <div
                className="col-lg-12 col-md-12 col-xs-12 grid p-0 business-service m-bot-28"
                key="2"
              >
                <div
                  className="service-description col-lg-12 col-md-12 col-xs-12 p-left-0"
                >
                  <h4
                    className="text-bold"
                  >
                    service2
                  </h4>
                  <p>
                    lorem ipsum...
                  </p>
                </div>
                <div
                  className="col-lg-12 col-md-12 col-xs-12 p-0 m-top-16"
                >
                  <Chip
                    canDelete={false}
                    key="54"
                    text="category1"
                  >
                    <a
                      className="
      search-filter-label
    "
                      data-value="category1"
                      onClick={[Function]}
                    >
                      <span
                        className="text"
                      >
                        category1
                      </span>
                    </a>
                  </Chip>
                </div>
              </div>
              <div
                className="col-lg-12 col-md-12 col-xs-12 grid p-0"
              >
                <div
                  className="col-lg-6 col-md-6 col-xs-6 p-0 m-right-54"
                >
                  <p
                    className="business-title"
                  >
                    Other Locations
                  </p>
                </div>
                <div
                  className="col-lg-4 col-md-4 col-xs-6 p-0"
                >
                  <p
                    className="business-title"
                  >
                    Contact:
                  </p>
                </div>
                <div
                  className="col-lg-12 col-md-12 col-xs-12 p-0 grid"
                >
                  <div
                    className="col-lg-12 col-md-12 col-xs-12 m-bot-24 p-0 grid"
                    key="2"
                  >
                    <div
                      className="col-lg-6 col-md-6 col-xs-6 p-0 m-right-54"
                    >
                      <h4>
                        456 Street Rd
                      </h4>
                      <h4>
                        Sacramento, CA 12345
                      </h4>
                    </div>
                    <div
                      className="col-lg-4 col-md-4 col-xs-6 p-0 m-top-24"
                    >
                      <h4>
                        (123) 456 - 7890
                      </h4>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <p
              className="location text-bold"
            >
              <span>
                2  Locations
              </span>
              <span
                className="m-x-7"
              >
                |
              </span>
              <span>
                Main location in 
                Sacramento
                , 
                CA
              </span>
            </p>
          </div>
        </div>
      </Business>
    </withRouter(Business)>
  </div>
</BusinessesList>
`;
